src/game_character.cpp

```diff
@@ -851,12 +850,8 @@ bool Game_Character::IsInPosition(int x, int y) const {
 	return ((GetX() == x) && (GetY() == y));
 }
 
-void Game_Character::SetOpacity(int opacity) {
-	this->opacity = min(max(0, opacity), 255);
-}
-
 int Game_Character::GetOpacity() const {
-	return opacity;
+	return (8 - GetTransparency()) * 32 - 1;
```

Comment:
@CherryDT Do you agree with this calculation?

Transparency is valued from 0 to 7, with 7 being almost completely transparent. This value can't go one more step to 8 to set fully transparent.

Reply:
Almost. It is a bit weirder. It is `255 - GetTransparency() * 30`. Yes, **30** and not 32.